docs/grammar/Solidity.g4

```diff
@@ -0,0 +1,489 @@
+	| whileStatement
+	| doWhileStatement
+	| continueStatement
+	| breakStatement
+	| tryStatement
+	| returnStatement
+	| emitStatement
+    | assemblyStatement
+;
+
+//@doc:inline
+simpleStatement: variableDeclarationStatement | expressionStatement;
+/**
+ * If statement with optional else part.
+ */
+ifStatement: If LParen expression RParen statement (Else statement)?;
+/**
+ * For statement with optional init, condition and post-loop part.
+ */
+forStatement: For LParen (simpleStatement | Semicolon) (expressionStatement | Semicolon) expression? RParen statement;
+whileStatement: While LParen expression RParen statement;
+doWhileStatement: Do statement While LParen expression RParen Semicolon;
+/**
+ * A continue statement. Only allowed inside for, while or do-while loops.
+ */
+continueStatement: Continue Semicolon;
+/**
+ * A break statement. Only allowed inside for, while or do-while loops.
+ */
+breakStatement: Break Semicolon;
+/**
+ * A try statement. The contained expression needs to be an external function call or a contract creation.
+ */
+tryStatement: Try expression (Returns LParen returnParameters=parameterList RParen)? block catchClause+;
+returnStatement: Return expression? Semicolon;
+/**
+ * An emit statement. The contained expression needs to refer to an event.
+ */
+emitStatement: Emit expression callArgumentList Semicolon;
+/**
+ * An inline assembly block.
+ * The contents of an inline assembly block use a separate scanner/lexer, i.e. the set of keywords and
+ * allowed identifiers is different inline an inline assembly block.
+ */
+assemblyStatement: Assembly AssemblyDialect? AssemblyLBrace yulStatement* YulRBrace;
+
+//@doc:inline
+variableDeclarationList: variableDeclarations+=variableDeclaration (Comma variableDeclarations+=variableDeclaration)*;
+/**
+ * A tuple of variable names to be used in variable declarations.
+ * May contain empty fields.
+ */
+variableDeclarationTuple:
+    LParen
+        (Comma* variableDeclarations+=variableDeclaration)
+        (Comma (variableDeclarations+=variableDeclaration)?)*
+    RParen;
+/**
+ * The declaration of one with an optional initial value or a tuple of variable with required initial value.
+ */
+variableDeclarationStatement: ((variableDeclaration (Assign expression)?) | (variableDeclarationTuple Assign expression)) Semicolon;
+expressionStatement: expression Semicolon;
+
+/**
+ * The catch clause of a try statement.
+ */
+catchClause: Catch (identifier? LParen (arguments=parameterList) RParen)? block;
+
+mappingType: Mapping LParen key=mappingKeyType Arrow value=typeName RParen;
+/**
+ * Only elementary types or user defined types are viable as mapping keys.
+ */
+mappingKeyType: elementaryTypeName[false] | userDefinedTypeName;
+
+/**
+ * A Yul statement within an inline assembly block.
+ * continue and break statements are only valid within for loops.
+ * leave statements are only valid within function bodies.
+ */
+yulStatement:
+    yulBlock
+	| yulVarDecl
```

Comment:
I'd use the same names as in `libyul/AsmData.h`, e.g. `yulVariableDeclaration` for this one.